base/rational.jl

```diff
@@ -11,12 +11,17 @@ struct Rational{T<:Integer} <: Real
     den::T
 
     function Rational{T}(num::Integer, den::Integer) where T<:Integer
-        num == den == zero(T) && __throw_rational_argerror(T)
+        num == den == zero(T) && __throw_rational_argerror_zero(T)
+        # issue #32569
+        if T <: Signed
+            signbit(den) && signbit(-den) && __throw_rational_argerror_typemin(T)
+        end
         num2, den2 = signbit(den) ? divgcd(-num, -den) : divgcd(num, den)
```

Comment:
As @KlausC suggested, this constructs unnecessarily `-den` once more than necessary. You can rearrange the code to fix that, e.g.
```julia
num2, den2 = if signbit(den)
    den = -den
    T <: Signed && signbit(den) && __throw_rational_argerror_typemin(T)
    divgcd(-num, den)
else
    divgcd(num, den)
end